config/katello.migrations/200605154320-dont-use-pulpcore-rpm-on-upgrades.rb

```diff
@@ -0,0 +1,9 @@
+PUPPET_CLASS = 'katello'.freeze
+PARAM = 'use_pulp_2_for_yum'.freeze
+
+def upgrade
+  answers[PUPPET_CLASS] = {} unless answers[PUPPET_CLASS].is_a?(Hash)
+  answers[PUPPET_CLASS][PARAM] = true
+end
+
+upgrade if answers[PUPPET_CLASS] && answers[PUPPET_CLASS][PARAM].nil?
```

Comment:
Is it that the value of that param would be nil or that the param exists when the migrations are executed? I can't seem to definitively recall if migrations take into account params from modules at the time they are executed or just act on the answers file as it exists without any view of params from the current set of modules.

Reply:
Based on:

 https://github.com/theforeman/kafo/blob/master/lib/kafo/configuration.rb#L48
https://github.com/theforeman/kafo/blob/master/lib/kafo/configuration.rb#L296-L298
https://github.com/theforeman/kafo/blob/master/lib/kafo/configuration.rb#L302

It looks like answers is just defined as the logical content of the parsed answers file, at the time that migrations are run.

So I think my implementation works, but we could be more confident if the tests would actually run, which they haven't. I'll try closing and re-opening the PR to see if that triggers it.